### PR TITLE
improve softmax perf

### PR DESCRIPTION
**Pull request by dding3**

### CODE REVIEW

**Comment by dding3** [comment]
jekins pass. http://10.112.231.51:18888/view/ZOO-PR/job/BigDL-PR-Dllib-Validation/106/

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala [hunk 1]
```diff
@@ -32,46 +33,17 @@ import scala.reflect.ClassTag
  * where shift = max_i(x_i).
  */
 @SerialVersionUID(- 7842335603491194236L)
-class SoftMax[T: ClassTag](var pos: Int = 1)(implicit ev: TensorNumeric[T])
+class SoftMax[T: ClassTag]()(implicit ev: TensorNumeric[T])
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala:36]: you should probably replace this class with `dllib.keras.layers.internal.InternalSoftMax`

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala [hunk 2]
```diff
@@ -81,13 +53,9 @@ class SoftMax[T: ClassTag](var pos: Int = 1)(implicit ev: TensorNumeric[T])
...
-    new SoftMax[T](pos)
-  }
-  def apply[@specialized(Float, Double) T: ClassTag]
+  def apply[@specialized(Float, Double) T: ClassTag]()
     (implicit ev: TensorNumeric[T]) : SoftMax[T] = {
-    new SoftMax[T](1)
+    new SoftMax[T]()
   }
   // Notice: SoftMin will call this function
   private[nn] def updateOutput[T: ClassTag](input: Tensor[T], output: Tensor[T],
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala:61]: are these two methods still used?

Comment (dding3) [reply scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala:61]: yes, they are used by Softmin

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala (hunk 1 above)
Comment (dding3) [reply scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala:36]: Ok. So we will move the InternalSoftMax implementation to nn.softmax and delete `InternalSoftMax` class?

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala (hunk 2 above)
Comment (jason-dai) [reply scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala:61]: then move them to SoftMin (maybe applying similar optimizations in a separate PR?)

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala (hunk 1 above)
Comment (jason-dai) [reply scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala:36]: yes

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala (hunk 2 above)
Comment (dding3) [reply scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala:61]: Do we need support softmin? Looks like tensorflow and keras doesn't support softmin.

Comment (jason-dai) [reply scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nn/SoftMax.scala:61]: We can then keep it as is

**Comment by jason-dai** [comment]
LGTM

**Comment by dding3** [comment]
jekins pass. http://10.112.231.51:18888/view/ZOO-PR/job/BigDL-PR-Dllib-Validation/123/
